Introduce autotest.sh for LDAP integrations tests, a minor fix and some …

blizzz opened the pull request:
…cleanup

I just run this as root (because docker):

```
$ sudo ./autotest.sh 
[sudo] password for foobar: 
./run-test.sh ./lib/user/DisplayName.php
f123287ace286f7ba40e33e3c56069322636236c4e3b50a0ffa566fd1786773a
aefb87c645d2630990dc2200ae477a17a54f40a84d2e61395ab2d7eb2bf42b27
CONTAINER ID        IMAGE                 COMMAND                 CREATED             STATUS                  PORTS                           NAMES
aefb87c645d2        osixia/phpldapadmin   "/container/tool/run"   1 seconds ago       Up Less than a second   80/tcp, 0.0.0.0:8443->443/tcp   docker-phpldapadmin
f123287ace28        nickstenning/slapd    "/sbin/my_init"         2 seconds ago       Up 1 seconds            127.0.0.1:7770->389/tcp         docker-slapd

LDAP server now available under 127.0.0.1:7770 (internal IP is 172.20.0.2)
phpldapadmin now available under https://127.0.0.1:8443

created user : Alice Celai
created user : Boris Orsib
running case1 
running case2 
Tests succeeded
Stopping and resetting containers
docker-slapd
docker-phpldapadmin
docker-slapd
docker-phpldapadmin
./run-test.sh ./lib/user/Avatar.php
5ef9da55b315e55ce10ef7d3bcdfde57a3014c101f6ff92c883206b70bd58a25
b2ea7508de36547e883cf93a1edfe269fcda110daf2b9e2a5791704caa7fd16b
CONTAINER ID        IMAGE                 COMMAND                 CREATED             STATUS                  PORTS                           NAMES
b2ea7508de36        osixia/phpldapadmin   "/container/tool/run"   1 seconds ago       Up Less than a second   80/tcp, 0.0.0.0:8443->443/tcp   docker-phpldapadmin
5ef9da55b315        nickstenning/slapd    "/sbin/my_init"         2 seconds ago       Up 1 seconds            127.0.0.1:7770->389/tcp         docker-slapd

LDAP server now available under 127.0.0.1:7770 (internal IP is 172.20.0.2)
phpldapadmin now available under https://127.0.0.1:8443

created user : Alice Eialc
created user : Boris Sriob
running case1 
running case2 
Tests succeeded
Stopping and resetting containers
docker-slapd
docker-phpldapadmin
docker-slapd
docker-phpldapadmin
./run-test.sh ./lib/FetchUsersByLoginName.php
f41cb2cd718a3f4788284d10ccbeee39ba0325a35519977b564c6efd5f598a52
28c11262ed1fcf5992f921d3ee0cd6790ff74bf4ed4db115e1f4c7371b88416a
CONTAINER ID        IMAGE                 COMMAND                 CREATED             STATUS                  PORTS                           NAMES
28c11262ed1f        osixia/phpldapadmin   "/container/tool/run"   1 seconds ago       Up Less than a second   80/tcp, 0.0.0.0:8443->443/tcp   docker-phpldapadmin
f41cb2cd718a        nickstenning/slapd    "/sbin/my_init"         2 seconds ago       Up 1 seconds            127.0.0.1:7770->389/tcp         docker-slapd

LDAP server now available under 127.0.0.1:7770 (internal IP is 172.20.0.2)
phpldapadmin now available under https://127.0.0.1:8443

created user : Alice Aleci
created user : Boris Iobrs
running case1 
running case2 
Tests succeeded
Stopping and resetting containers
docker-slapd
docker-phpldapadmin
docker-slapd
docker-phpldapadmin
./run-test.sh ./lib/CountUsersByLoginName.php
9a4e18d7eba2db0cb904a30fabe9d605a49f071da0827573143f3a41e9f20646
a37fb16b5c07201674769bcf81a73a463b90780188599c0b8a9a4877abc97b08
CONTAINER ID        IMAGE                 COMMAND                 CREATED             STATUS                  PORTS                           NAMES
a37fb16b5c07        osixia/phpldapadmin   "/container/tool/run"   1 seconds ago       Up Less than a second   80/tcp, 0.0.0.0:8443->443/tcp   docker-phpldapadmin
9a4e18d7eba2        nickstenning/slapd    "/sbin/my_init"         1 seconds ago       Up 1 seconds            127.0.0.1:7770->389/tcp         docker-slapd

LDAP server now available under 127.0.0.1:7770 (internal IP is 172.20.0.2)
phpldapadmin now available under https://127.0.0.1:8443

created user : Alice Eicla
created user : Boris Siobr
running case1 
running case2 
Tests succeeded
Stopping and resetting containers
docker-slapd
docker-phpldapadmin
docker-slapd
docker-phpldapadmin
./run-test.sh ./lib/Connect.php
82a3fd8aa5b40b42c97a56f2d66b6275d1a7ddf8118a4f638e330630ee84cb13
18bfb0bb29267e244403191a48837a2e893000fdf516bb648d64fd9725fd6221
CONTAINER ID        IMAGE                 COMMAND                 CREATED             STATUS                  PORTS                           NAMES
18bfb0bb2926        osixia/phpldapadmin   "/container/tool/run"   1 seconds ago       Up Less than a second   80/tcp, 0.0.0.0:8443->443/tcp   docker-phpldapadmin
82a3fd8aa5b4        nickstenning/slapd    "/sbin/my_init"         2 seconds ago       Up Less than a second   127.0.0.1:7770->389/tcp         docker-slapd

LDAP server now available under 127.0.0.1:7770 (internal IP is 172.20.0.2)
phpldapadmin now available under https://127.0.0.1:8443

running case1 
running case2 
running case3 
running case4 
running case5 
running case6 
Tests succeeded
Stopping and resetting containers
docker-slapd
docker-phpldapadmin
docker-slapd
docker-phpldapadmin
./run-test.sh ./lib/AccessGroupsMatchFilter.php
f1a6806e08bd3229ec3a35f5a5c4a13c893d6b0a6b2d234a9586738022717525
980d8611fc749d654567bcb1133c7446f5d9765ede83bd040863120f08bc957b
CONTAINER ID        IMAGE                 COMMAND                 CREATED             STATUS                  PORTS                           NAMES
980d8611fc74        osixia/phpldapadmin   "/container/tool/run"   1 seconds ago       Up Less than a second   80/tcp, 0.0.0.0:8443->443/tcp   docker-phpldapadmin
f1a6806e08bd        nickstenning/slapd    "/sbin/my_init"         2 seconds ago       Up 1 seconds            127.0.0.1:7770->389/tcp         docker-slapd

LDAP server now available under 127.0.0.1:7770 (internal IP is 172.20.0.2)
phpldapadmin now available under https://127.0.0.1:8443

created user : Alice Ielac
created user : Boris Ibors
created group : RedGroup
created group : BlueGroup
created group : GreenGroup
created group : PurpleGroup
created group : SquareGroup
created group : CircleGroup
created group : TriangleGroup
created group : SquaredCircleGroup
running case1 
running case2 
running case3 
Tests succeeded
Stopping and resetting containers
docker-slapd
docker-phpldapadmin
docker-slapd
docker-phpldapadmin
./run-test.sh ./lib/BatchApplyUserAttributes.php
b62ba6e09d60f318c2338815ca0bf15c590329a66b327b785cfbf9a534fd1c19
0580a429c3c028321cf24e5b5f7da82a0a294edca5f20f04472351d5bd10bb06
CONTAINER ID        IMAGE                 COMMAND                 CREATED             STATUS                  PORTS                           NAMES
0580a429c3c0        osixia/phpldapadmin   "/container/tool/run"   1 seconds ago       Up Less than a second   80/tcp, 0.0.0.0:8443->443/tcp   docker-phpldapadmin
b62ba6e09d60        nickstenning/slapd    "/sbin/my_init"         2 seconds ago       Up 1 seconds            127.0.0.1:7770->389/tcp         docker-slapd

LDAP server now available under 127.0.0.1:7770 (internal IP is 172.20.0.2)
phpldapadmin now available under https://127.0.0.1:8443

created user : Alice Alcei
created user : Boris Ibrso
created user : Robot
running case1 
Tests succeeded
Stopping and resetting containers
docker-slapd
docker-phpldapadmin
docker-slapd
docker-phpldapadmin
./run-test.sh ./lib/BackupServer.php
139ac287bf5a382fb13ce8fe9e1d21180b3a3fbbe644e554dcd5fc71439cf814
77a8dc79b676266857db2ac4d45b1865d1625e19140e1631349971eea8190138
CONTAINER ID        IMAGE                 COMMAND                 CREATED             STATUS                  PORTS                           NAMES
77a8dc79b676        osixia/phpldapadmin   "/container/tool/run"   1 seconds ago       Up Less than a second   80/tcp, 0.0.0.0:8443->443/tcp   docker-phpldapadmin
139ac287bf5a        nickstenning/slapd    "/sbin/my_init"         2 seconds ago       Up 1 seconds            127.0.0.1:7770->389/tcp         docker-slapd

LDAP server now available under 127.0.0.1:7770 (internal IP is 172.20.0.2)
phpldapadmin now available under https://127.0.0.1:8443

running case1 
running case2 
running case3 
Tests succeeded
Stopping and resetting containers
docker-slapd
docker-phpldapadmin
docker-slapd
docker-phpldapadmin
./run-test.sh ./lib/UserHome.php
6cca1efa5d67eb4aa68c02d02840c51b257297862d650d14c55cc35925d53664
95d75caf407acf49b04c3320f5fe14035436eadb9dc4155f94752dcf9f980d02
CONTAINER ID        IMAGE                 COMMAND                 CREATED             STATUS                  PORTS                           NAMES
95d75caf407a        osixia/phpldapadmin   "/container/tool/run"   1 seconds ago       Up Less than a second   80/tcp, 0.0.0.0:8443->443/tcp   docker-phpldapadmin
6cca1efa5d67        nickstenning/slapd    "/sbin/my_init"         2 seconds ago       Up 1 seconds            127.0.0.1:7770->389/tcp         docker-slapd

LDAP server now available under 127.0.0.1:7770 (internal IP is 172.20.0.2)
phpldapadmin now available under https://127.0.0.1:8443

created user : Alice Ecila
created user : Boris Bsiro
running case1 
running case2 
running case3 
Tests succeeded
Stopping and resetting containers
docker-slapd
docker-phpldapadmin
docker-slapd
docker-phpldapadmin
./run-test.sh ./lib/Paging.php
562e8cac3fb59005aa7ac7dfbbf613465d35bfce4ab0631c231218c267feb33d
c8780c1c79ef907c3d3feaf42915e007f879a03025331f57c146e3ded228447c
CONTAINER ID        IMAGE                 COMMAND                 CREATED             STATUS                  PORTS                           NAMES
c8780c1c79ef        osixia/phpldapadmin   "/container/tool/run"   1 seconds ago       Up Less than a second   80/tcp, 0.0.0.0:8443->443/tcp   docker-phpldapadmin
562e8cac3fb5        nickstenning/slapd    "/sbin/my_init"         2 seconds ago       Up 1 seconds            127.0.0.1:7770->389/tcp         docker-slapd

LDAP server now available under 127.0.0.1:7770 (internal IP is 172.20.0.2)
phpldapadmin now available under https://127.0.0.1:8443

created user : Alice Laeic
created user : Boris Biosr
running case1 
Tests succeeded
Stopping and resetting containers
docker-slapd
docker-phpldapadmin
docker-slapd
docker-phpldapadmin
```

It would stop after the first test that fails.

@DeepDiver1975 @MorrisJobke

blizzz committed:
Introduce autotest.sh for LDAP integrations tests, a minor, and some cleanup

MorrisJobke commented:
```
morrisjobke@gymir:~/Projects/owncloud/master/apps/user_ldap/tests/integration(ldap-integration-tests-1up○) » ./autotest.sh 
./run-test.sh ./lib//AccessGroupsMatchFilter.php
./run-test.sh: line 16: setup-scripts/start.sh: No such file or directory
./run-test.sh: line 18: setup-scripts/stop.sh: No such file or directory
morrisjobke@gymir:~/Projects/owncloud/master/apps/user_ldap/tests/integration(ldap-integration-tests-1up○) » ls -la                                                                                                                        127 ↵
total 56
drwxr-xr-x  11 morrisjobke  staff   374  7 Apr 12:03 .
drwxr-xr-x  13 morrisjobke  staff   442  7 Apr 10:12 ..
-rw-r--r--   1 morrisjobke  staff  3455  7 Apr 10:12 abstractintegrationtest.php
-rwxr-xr-x   1 morrisjobke  staff   191  7 Apr 12:03 autotest.sh
drwxr-xr-x   4 morrisjobke  staff   136 24 Mär 13:17 data
-rw-r--r--   1 morrisjobke  staff  5272  7 Apr 10:12 exceptionOnLostConnection.php
-rw-r--r--   1 morrisjobke  staff  1216  7 Apr 10:12 fakemanager.php
drwxr-xr-x  11 morrisjobke  staff   374  7 Apr 12:03 lib
-rw-r--r--   1 morrisjobke  staff  2160 24 Mär 13:17 readme.md
-rwxr-xr-x   1 morrisjobke  staff   494  7 Apr 12:03 run-test.sh
drwxr-xr-x   6 morrisjobke  staff   204  7 Apr 12:03 setup-scripts
morrisjobke@gymir:~/Projects/owncloud/master/apps/user_ldap/tests/integration(ldap-integration-tests-1up○) » ls setup-scripts 
createExplicitGroups.php            createExplicitGroupsDifferentOU.php createExplicitUsers.php             createUsersWithoutDisplayName.php
```

blizzz commented:
@MorrisJobke see the readme file

MorrisJobke commented:
> Configure config.php according to your needs, also have a look into the LDAP and network settings in start.sh and stop.sh.

🙈

DeepDiver1975 commented:
@blizzz I like this - I'm willing to setup this on jenkins as soon as this is fully automated by running one script.

blizzz commented:
@DeepDiver1975 Roger, I'll tune autotest.sh to get the additional files and create a default config

* [ ] fetch start.sh and stop.sh if missing
* [ ] create config.php if missing
* [ ] optinal: pass config per arguments other than hardcoding them. If necessary.

blizzz committed:
Fetch start.sh, stop.sh and config.php if files are missing

blizzz committed:
Keep var name as is, comes from extenal config.php

blizzz committed:
Update readme

blizzz commented:
@DeepDiver1975 @MorrisJobke please have a fresh look. Requires wget to fetch files.

MorrisJobke commented:
Funny ... it starts a lot of containers, prints no errors but also doesn't list any tests. The return value is 0.

```
...
./run-test.sh ./lib//user/DisplayName.php
Using default tag: latest
latest: Pulling from nickstenning/slapd
d634beec75db: Already exists 
8b877d3adb63: Already exists 
95dde7ac21e8: Already exists 
51cd589d7fd2: Already exists 
7a4944404752: Already exists 
06220fbcea39: Already exists 
2bb8bc4a46bc: Already exists 
0127e1e850d2: Already exists 
c604a0c39d93: Already exists 
c590d12a3d65: Already exists 
69413d713fa3: Already exists 
39385bb7e2ba: Already exists 
e733655921b5: Already exists 
f7044d0db20a: Already exists 
b77a39f46e2b: Already exists 
20983c17a2db: Already exists 
b442383b3bde: Already exists 
6a4eb7d6ca9e: Already exists 
e4778dfe2bf8: Already exists 
6ccecf564695: Already exists 
7767fc2133ea: Already exists 
094465639d0e: Already exists 
b18d57ced551: Already exists 
Digest: sha256:5227ee177df20e62252651aaaff870b505b6cae1a1a2127fca56ece8cbaa3874
Status: Image is up to date for nickstenning/slapd:latest
71a8706bd2709cf4ed0c7090e4787150a937e29f80aa1bd66ee68c77739bf1bf
Using default tag: latest
latest: Pulling from osixia/phpldapadmin
60f618e4610b: Already exists 
84a94891c2fa: Already exists 
8e0d41a4a93c: Already exists 
39be7a5c4d01: Already exists 
ed3d7675c948: Already exists 
053f001cede3: Already exists 
7d50e45aa4fb: Already exists 
965f8d79fa50: Already exists 
80ca957fc32c: Already exists 
e8cfde3a1fb0: Already exists 
826b87c88e92: Already exists 
c47cbf82e3f0: Already exists 
b4b457bffb02: Already exists 
c8a73e719394: Already exists 
fa57433e802d: Already exists 
39b6f40e3992: Already exists 
2846a6b12ab1: Already exists 
8187c6ac41b9: Already exists 
Digest: sha256:79e9d2469aa2a0cd4b53a8c63451ead47b2ec0260dd6208919ba7c8fff5ceb10
Status: Image is up to date for osixia/phpldapadmin:latest
651365737e1581412fb66957f95d4f7417131111c4676d3c978bca29f941271c
CONTAINER ID        IMAGE                 COMMAND                 CREATED                  STATUS                  PORTS                           NAMES
651365737e15        osixia/phpldapadmin   "/container/tool/run"   Less than a second ago   Up Less than a second   80/tcp, 0.0.0.0:8443->443/tcp   docker-phpldapadmin
71a8706bd270        nickstenning/slapd    "/sbin/my_init"         3 seconds ago            Up 2 seconds            127.0.0.1:389->389/tcp          docker-slapd

LDAP server now available under 127.0.0.1:389 (internal IP is 172.17.0.28)
phpldapadmin now available under https://127.0.0.1:8443

Stopping and resetting containers
docker-slapd
docker-phpldapadmin
docker-slapd
docker-phpldapadmin
./run-test.sh ./lib//UserHome.php
Using default tag: latest
latest: Pulling from nickstenning/slapd
d634beec75db: Already exists 
8b877d3adb63: Already exists 
95dde7ac21e8: Already exists 
51cd589d7fd2: Already exists 
7a4944404752: Already exists 
06220fbcea39: Already exists 
2bb8bc4a46bc: Already exists 
0127e1e850d2: Already exists 
c604a0c39d93: Already exists 
c590d12a3d65: Already exists 
69413d713fa3: Already exists 
39385bb7e2ba: Already exists 
e733655921b5: Already exists 
f7044d0db20a: Already exists 
b77a39f46e2b: Already exists 
20983c17a2db: Already exists 
b442383b3bde: Already exists 
6a4eb7d6ca9e: Already exists 
e4778dfe2bf8: Already exists 
6ccecf564695: Already exists 
7767fc2133ea: Already exists 
094465639d0e: Already exists 
b18d57ced551: Already exists 
Digest: sha256:5227ee177df20e62252651aaaff870b505b6cae1a1a2127fca56ece8cbaa3874
Status: Image is up to date for nickstenning/slapd:latest
4d73463e30747e6b1a6cc04920e6ac66ccad611d786a0b3b2e7b34c6ffc72365
Using default tag: latest
latest: Pulling from osixia/phpldapadmin
60f618e4610b: Already exists 
84a94891c2fa: Already exists 
8e0d41a4a93c: Already exists 
39be7a5c4d01: Already exists 
ed3d7675c948: Already exists 
053f001cede3: Already exists 
7d50e45aa4fb: Already exists 
965f8d79fa50: Already exists 
80ca957fc32c: Already exists 
e8cfde3a1fb0: Already exists 
826b87c88e92: Already exists 
c47cbf82e3f0: Already exists 
b4b457bffb02: Already exists 
c8a73e719394: Already exists 
fa57433e802d: Already exists 
39b6f40e3992: Already exists 
2846a6b12ab1: Already exists 
8187c6ac41b9: Already exists 
Digest: sha256:79e9d2469aa2a0cd4b53a8c63451ead47b2ec0260dd6208919ba7c8fff5ceb10
Status: Image is up to date for osixia/phpldapadmin:latest
e38e58e52dc44dad1d1cddc66fad2a8512409057c673e237db423ad8f2cdeb19
CONTAINER ID        IMAGE                 COMMAND                 CREATED             STATUS                  PORTS                           NAMES
e38e58e52dc4        osixia/phpldapadmin   "/container/tool/run"   1 seconds ago       Up Less than a second   80/tcp, 0.0.0.0:8443->443/tcp   docker-phpldapadmin
4d73463e3074        nickstenning/slapd    "/sbin/my_init"         4 seconds ago       Up 2 seconds            127.0.0.1:389->389/tcp          docker-slapd

LDAP server now available under 127.0.0.1:389 (internal IP is 172.17.0.30)
phpldapadmin now available under https://127.0.0.1:8443

Stopping and resetting containers
docker-slapd
docker-phpldapadmin
docker-slapd
docker-phpldapadmin
```

MorrisJobke commented:
Is this intended? Could we make the pull quiet?

blizzz commented:
No, that's strange, I give it another look.

MorrisJobke commented:
> Could we make the pull quiet?

https://github.com/owncloud/administration/pull/75 This makes the output less verbose and errors are still shown.

MorrisJobke commented:
And please add following three files to the gitignore ;)

```
	setup-scripts/config.php
	setup-scripts/start.sh
	setup-scripts/stop.sh
```

blizzz committed:
Git-Ignore setup-scripts originating from administration-repo

blizzz committed:
decrease docker and wget verbosity

blizzz commented:
@MorrisJobke 

> owncloud/administration#75 This makes the output less verbose and errors are still shown.

which errors are shown?

I silenced it down some more, including wget, for me it now looks like:

```
$ sudo apps/user_ldap/tests/integration/autotest.sh                
fetching https://raw.githubusercontent.com/owncloud/administration/master/ldap-testing/start.sh
fetching https://raw.githubusercontent.com/owncloud/administration/master/ldap-testing/stop.sh
./run-test.sh ./lib/user/DisplayName.php
created user : Alice Lecia
created user : Boris Rsboi
running case1 
running case2 
Tests succeeded

./run-test.sh ./lib/user/Avatar.php
created user : Alice Eical
created user : Boris Isrbo
running case1 
running case2 
Tests succeeded

./run-test.sh ./lib/FetchUsersByLoginName.php
created user : Alice Ealic
created user : Boris Sriob
running case1 
running case2 
Tests succeeded

./run-test.sh ./lib/CountUsersByLoginName.php
…
```

Mind to retry?

MorrisJobke commented:
> which errors are shown?

I just tested with a mistyped image name. So no error during the autotest.sh run.

Maybe its because I use a non-conventional setup, where the ports are not forwarded to localhost, but the VM, where the docker is running. (on OS X I need to use this)

But the output looks weird:

```
./autotest.sh
./run-test.sh ./lib//AccessGroupsMatchFilter.php

./run-test.sh ./lib//BackupServer.php

./run-test.sh ./lib//BatchApplyUserAttributes.php

./run-test.sh ./lib//Connect.php

./run-test.sh ./lib//CountUsersByLoginName.php

./run-test.sh ./lib//FetchUsersByLoginName.php

./run-test.sh ./lib//Paging.php

./run-test.sh ./lib//user/Avatar.php

./run-test.sh ./lib//user/DisplayName.php

./run-test.sh ./lib//UserHome.php

```

Maybe the error handling needs to be improved

blizzz commented:
@MorrisJobke that's still not how it should look like. Compare with my output.

blizzz committed:
enable user_ldap on test run, and execute scripts as web user (config file owner)

MorrisJobke commented:
> @MorrisJobke that's still not how it should look like. Compare with my output.

Looks like a setup issue on the one side and an error handling issue on the other.

blizzz commented:
We did a debug session:

* [ ] BUG (possible): user_ldap should be ensured to be enabled
* [ ] BUG: PHP errors are logged away in the owncloud.log file, so it returns exit code 0 instead of an expected > 0. This looks like tests run successfully.

blizzz committed:
On exit code 0, ensure that test run successfully.

It might be just as well, that ownCloud caught an error making php exit
with 0.

blizzz commented:
@MorrisJobke the other issues are addressed. When user_ldap is not enabled for instance (which won't happen anymore, I temporarily added a disable-line for testing) it looks like this:

```
$ sudo ./autotest.sh 
./run-test.sh ./lib/user/DisplayName.php
user_ldap enabled
user_ldap disabled

a superordinate error occurred – please check the owncloud.log
$ echo $?
1
```

MorrisJobke commented:
I still sttruggling with my setup, but the error code seems to be not properly forwarded in the case the wrong LDAP host is set:

```
$ ./run-test.sh ./lib//AccessGroupsMatchFilter.php
Can't contact LDAP server%                                                                                                                                                                   $ echo $?
0
```

DeepDiver1975 commented on .gitignore:
```diff
@@ -31,6 +31,11 @@
 /apps/files_external/tests/config.*.php
 
 
+# ignore integration test-related, external files in user_ldap
```
I'd keep them in here - especially thinking about the stable branches all referring to administration master

MorrisJobke commented:
> I'm still sttruggling with my setup, but the error code seems to be not properly forwarded in the case the wrong LDAP host is set:

This https://github.com/owncloud/core/blob/8b9bcbdf95add986d18bd9e29fb7c6349c8d7a62/apps/user_ldap/tests/integration/lib/AccessGroupsMatchFilter.php#L35-L35 fails but didn't give back the proper return code. I will debug further.

MorrisJobke committed:
return proper error codes

MorrisJobke commented on apps/user_ldap/tests/integration/run-test.sh:
```diff
@@ -13,9 +13,19 @@ fi
+    # so we add an extra check whether tests really succeeded
+    if [[ $OUT != *"Tests succeeded"* ]]; then
+        echo "a superordinate error occurred – please check the owncloud.log"
+        CODE=1
```
I fixed this in a less hacky way: 29e7766

MorrisJobke commented on apps/user_ldap/tests/integration/run-test.sh:
```diff
@@ -11,9 +11,11 @@ if [ ! -e "$TESTSCRIPT" ] ; then
-
 # sleep is necessary, otherwise the LDAP server cannot be connected to, yet.
-setup-scripts/start.sh > /dev/null && sleep 5 && php -f "$TESTSCRIPT"
+WEBUSER=`stat -c %U ../../../../config/config.php`
```
On OS X there is the BSD stat and not the GNU one -.-

We have something similar in console.php. Let me check.

MorrisJobke replied:
I fixed this too.

MorrisJobke committed:
make run-test.sh POSIX compliant

MorrisJobke commented:
Got it ... you bound the LDAP port of the docker container to localhost. Also the name should not be hard coded to allow to run multiple of these tests in parallel. We should simply forward this to a random port and then propagate this random port to the config.php. We already did this for the files_external tests. (there we directly used the internal IP address of the container - for the LDAP I would use the random port method)

I would also vote for integration the start and stop scripts to core and adjust them to allow the propagation of the port. I could help you here if you want :)

blizzz commented:
Thanks for the commits :fireworks:

blizzz commented:
> Got it ... you bound the LDAP port of the docker container to localhost.

I did? where?

MorrisJobke commented:
> I did? where?

https://github.com/owncloud/administration/blob/master/ldap-testing/start.sh#L17

blizzz commented:
>> I did? where?

> https://github.com/owncloud/administration/blob/master/ldap-testing/start.sh#L17

Ah… well, well, I did not get a single commit inside that file :tongue: ;)

MorrisJobke commented:
> Ah… well, well, I did not get a single commit inside that file  ;)

Ah ... it was Vincent :D No worries :) I managed to fix it.

blizzz commented:
I did not retest yet, but as far as i can see basically

* [ ] https://github.com/owncloud/core/pull/23832#discussion-diff-61218508

needs to be dealt with.

cc @SergioBertolinSG

PVince81 commented:
Needs rebase. Massive conflicts due to the move to PSR-4 :frowning: 

CC @nickvergessen if you can help somehow

nickvergessen commented:
I can try to do the rebase, but as far as I can see, there are still open comments.

nickvergessen commented:
Rebased (without addressing additional comments)

PVince81 commented:
Moving this to 9.2

PVince81 commented:
@DeepDiver1975 something to move and reuse in the user_ldap repo ?

PVince81 commented:
@jvillafanez move to the ldap repo ?

jvillafanez commented:
Move to https://github.com/owncloud/user_ldap/pull/17
